src/main/java/duke/Event.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static duke.DukeConstants.KEY_SEPARATOR;
+
+/**
+ * This class is used to construct an Event task.
+ */
+public class Event extends Task {
+    private LocalDate at;
+    private String time;
+
+    /**
+     * Constructor for the Event class.
+     *
+     * @param description Task description.
+     * @param at Date of task.
+     */
+    public Event(String description, LocalDate at) {
+        super(description);
+        this.at = at;
+        this.time = "";
+    }
+
+    /**
+     * Constructor for the Event class.
+     *
+     * @param description Task description.
+     * @param at Date of task.
+     * @param time Time of task.
+     */
+    public Event(String description, LocalDate at, String time) {
+        super(description);
+        this.at = at;
+        this.time = time;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String formatToSave() {
```

Comment:
Perhaps can have a naming leaning more towards action verb.

Reply:
thanks for your comments in the other parts of the code! however im pretty sure format is a verb haha :]